frontend/tests/unit/models/trick.test.js

```diff
@@ -205,4 +205,139 @@ describe("extras", () => {
 
     expect(trick.extras()).toEqual([extras.fox]);
   });
+
+  test("should see charlie in random trick", () => {
+    const trick = new Trick(4);
+
+    player1.isRe = () => true;
+    player2.isRe = () => true;
+    player3.isRe = () => false;
+    player4.isRe = () => false;
+
+    trick.add(jack.of(suits.hearts), player3);
+    trick.add(jack.of(suits.clubs), player4);
+    trick.add(ten.of(suits.hearts), player1);
+    trick.add(queen.of(suits.spades), player2);
+
+    expect(trick.extras()).toEqual([]);
+  });
+
+  test("should catch charlie", () => {
+    const trick = new Trick(4);
+    trick.setLastTrickInRound();
+
+    player1.isRe = () => true;
+    player2.isRe = () => true;
+    player3.isRe = () => false;
+    player4.isRe = () => false;
+
+    trick.add(jack.of(suits.hearts), player3);
+    trick.add(jack.of(suits.clubs), player4);
+    trick.add(ten.of(suits.hearts), player1);
+    trick.add(queen.of(suits.spades), player2);
+
+    expect(trick.extras()).toEqual([extras.charlie_caught]);
+  });
+
+  test("should catch two charlies", () => {
+    const trick = new Trick(4);
+    trick.setLastTrickInRound();
+
+    player1.isRe = () => true;
+    player2.isRe = () => true;
+    player3.isRe = () => false;
+    player4.isRe = () => false;
+
+    trick.add(jack.of(suits.clubs), player3);
+    trick.add(jack.of(suits.clubs), player4);
+    trick.add(ten.of(suits.hearts), player1);
+    trick.add(queen.of(suits.spades), player2);
+
+    expect(trick.extras()).toEqual([extras.charlie_caught, extras.charlie_caught]);
+  });
+
+  test("should see charlie, nothing happens", () => {
+    const trick = new Trick(4);
+    trick.setLastTrickInRound();
+
+    player1.isRe = () => true;
+    player2.isRe = () => true;
+    player3.isRe = () => false;
+    player4.isRe = () => false;
```

Comment:
I think we only need to define this for the `fox caught` and `charlie caught` cases and can leave it out for others? makes tests a little shorter. Maybe we simply add it to a `beforeEach()` block in the `describe("extras", () => {` block?

```js
describe("extras", () => {
  beforeEach(() => {
    player1.isRe = () => true;
    player2.isRe = () => true;
    player3.isRe = () => false;
    player4.isRe = () => false;
  });

  // tests here
```

Reply:
I have no clue if this will be working, but I'm happy to implement and try 😁